simnode/generator.py

```diff
@@ -85,50 +87,166 @@ def _calculate_parallel_coords(offset: float, line_width: float) \
+                #  should the older method be integrated? (by simulating overloading using type()?)
+                #  The functionality is basically the same, except it works for changing widths and needs a list as
+                #  input. I tried to make the operation similar.
+                def _calculate_parallel_coords_list(offset: list, line_width: float) \
+                        -> Optional[List[Tuple[float, float, float, float]]]:
+                    """ calculates parallel coordinates of a road
+
+                    :param offset: list of offsets, must be smaller than num_nodes//2, should be equidistant
+                    :param line_width: specifies the width of the output coordinates
+                    :return: coordinates for the road
+                    """
+
+                    assert offset.__len__() < self.add_roads_to_scenario.num_nodes//2, \
+                        "there have to be less than half offset points of num_node for shapely LineStrings to work"
+
+                    original_line = LineString(zip(new_x_vals, new_y_vals))
+                    # num_points = original_line.length # seems to sometimes be trimmed to a bit more than 100?
+                    num_points = self.add_roads_to_scenario.num_nodes
+                    cutting_points = linspace(0, num_points-1, dtype=int, num=min(num_points, offset.__len__()))
+                    # extend the last point just a bit to get all nodes
+                    cutting_points[-1] = cutting_points[-1] + cutting_points [-1] - cutting_points[-2]
+
+                    offset_sub_lines = ()
+                    previous_p = 1
+                    i = 0
+                    # split road into multiple pieces
+                    for p in cutting_points:
+                        # cython int32 to int
+                        p = int(p)
+                        if p > 0:
+                            try:
+                                # calculate parallel offset for each road piece
+                                road_lnstr = LineString(original_line.coords[previous_p: p]).parallel_offset(offset[i])
+                                if offset_sub_lines.__len__() == 0:
+                                    offset_sub_lines = road_lnstr.coords.xy
+                                    # needs to be reversed for positive offset (why?)
+                                    if offset[i] > 0:
+                                        offset_sub_lines[0].reverse()
+                                        offset_sub_lines[1].reverse()
+                                else:
+                                    # try except not pretty, but problems with empty line string segments solved
+                                    try:
+                                        line_str_0 = (road_lnstr.coords.xy)[0]
+                                        line_str_1 = (road_lnstr.coords.xy)[1]
+                                        # needs to be reversed for positive offset (why?)
```

Comment:
That's an interesting question. At this point it would be very helpful to explain why it is like that.